File: MyMindNotes/app/src/main/java/com/android/mymindnotes/retrofit/CheckNicknameApi.java
package com.android.mymindnotes.retrofit;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface CheckNicknameApi {
    @GET("/api/member/checkNickname/{nickname}")
    Call<Map<String, Object>> checkNickname(@Path("nickname") String nickname);
}

File: MyMindNotes/data/src/main/java/com/android/mymindnotes/data/retrofit/api/user/CheckNickNameApi.kt
package com.android.mymindnotes.data.retrofit.api.user

import com.android.mymindnotes.core.dto.DuplicateCheckResponse
import retrofit2.http.GET
import retrofit2.http.Path

interface CheckNickNameApi {
    @GET("/api/member/checkNickname/{nickname}")
    suspend fun checkNickname(
        @Path("nickname") nickname: String
    ): DuplicateCheckResponse
}




